website/content/docs/configure.mdx

```diff
@@ -1,87 +1,73 @@
 
-- On a Unix system, if the `$PACKER_CONFIG_DIR` environment variable is set to
-  `/home/packer`, the config directory will be: `/home/packer/.packer.d/` and
-  other values will not be checked.
-- On a Windows system, if the `PACKER_CONFIG_DIR` environment variable is set to `C:/`,the
-  config directory will be: `C:/packer.d/` and other values will not be checked.
 
 <a id="packer-s-config-file"></a>
 
-## Packer's config file (deprecated)
+##  Install a JSON configuration file
+
+This installation method is deprecated since 1.7. Instead, use the [HCL2
+`required_plugins` block](/packer/docs/templates/hcl_templates/blocks/packer#specifying-plugin-requirements) to configure how Packer installs plugins. 
 
-Packer can optionally read a JSON file for the end user to set core settings.
-The config file of Packer will be looked up on the following paths:
+You use a JSON file to set core Packer settings. Packer checks the following paths for the configuration file:
 
 | Unix                             | Windows                           |
 | -------------------------------- | --------------------------------- |
 | `${PACKER_CONFIG}`               | `%PACKER_CONFIG%`                 |
 | `${HOME}/.packerconfig`          | `%APPDATA%\packer.config\`        |
 
-The format of the configuration file is basic JSON.
 
-### Packer config file configuration Reference
+### JSON configuration file reference 
+
+Use basic JSON to configure the file. You can specify the configuration parameters for the core
+configuration file. All parameters are optional. None of these are required, since all have defaults.
 
-Below is the list of all available configuration parameters for the core
-configuration file. None of these are required, since all have defaults.
+- `plugin_min_port`: Number that specifies the lowest port that Packer can use for communicating with plugins. Packer communicates with plugins over TCP connections on your local host. Default is `10000`. We recommend setting a wide range between `plugin_min_port` and `plugin_max_port` so that Packer has access to at least 25 ports on a single run.
+- `plugin_max_port`: Number that specifies highest port that Packer can for communicating with plugins. Packer communicates with plugins over TCP connections on your local host. Default is `25000`. We recommend setting a wide range between `plugin_min_port` and `plugin_max_port` so that Packer has access to at least 25 ports on a single run.
+- `builders`: Specifies a type of object that installs plugins. Refer to [Builders](/packer/docs/builders) for additional information. 
+- `commands`: Specifies a type of object that installs plugins. 
+- `post-processors`: Specifies a type of object that installs plugins. Refer to [Post-Processors](/packer/docs/post-processors) for additional information.
+- `provisioners`: Specifies a type of object that installs plugins. Refer to [Provisioners](/packer/docs/provisioners) for additional information. 
```

Comment:
Is this comment directed at the statement about the `packer init` command in line 60? Are you suggesting that we remove it?

Reply:
Not related to `packer init` per se, but `config.json` (the legacy config file) used to make it possible to declare which plugins were required by the template. That was JSON-only, and only for declaring mono-component plugins usage.

Since we don't discover them anymore on other parts of the code, and the config file is deprecated anyway, I would indeed suggest we remove references to components here. So only document the general configuration options, and leave the plugin installation/discovery to the modern alternatives.

I'm building a PR that will remove that path and explicitly error if they're encountered from the config file, so users will know these won't be used at all.
Of course this implementation may change by the time we release, depending on @nywilken's feedback then when he's back into office.

Reply:
https://github.com/hashicorp/packer/pull/12998
^ The related PR here

Reply:
OK, just to be clear, we should remove the entire `## Install a JSON configuration file` section, correct? Or are you saying that we also need to remove the `## Install a configuration file` section, too? In effect, the only way to configure Packer is to export the environment variables described in `## Configuration reference`?

Reply:
I imagine `commands` is a leftover from an old spec, I don't see it in the structure we deserialise into

Reply:
Should we remove this section entierly then given this is out-of-date, or should we do it in a subsquent PR?